### PR TITLE
LP-2431 Implement CI/CD pipeline for unit tests

### PR DESCRIPTION
**Pull request by zahid-arbisoft**
### Implement CI/CD pipeline for unit tests

- [x] Circleci job will run for each commit on PR
- [x] For PRs against the develop branch, only tests for ADG will run, and corresponding code coverage of `openedx/adg/*` will be posted to PR.
- [x] diff-cover will show coverage on only changes files/lines on PR as compared to PR base (target) branch
- [x] For PRs against the master branch, all tests will run with coverage
- [x] Job will fail if coverage is less than 80%
- [x] Any test can be skipped, by adding it in `TEST_SKIP_LIST` at `edx-platform/adg_pipelines/skip_tests.py`
- [x] Bash script `edx-run-test.sh` can be used to run ADG test with coverage on local machine
- [x] Django requirements and OS dependencies will be cached
- [x] All test reports will be uploaded to circleci as artifacts
- [x] Failing test with error will appear or circleci
- [x] Codecov integrated
- [x] Utilities moved from `openedx/adg/common/util` to `openedx/adg/lms/utils`
- [ ] Core changes in 2 files will be logged after PR review

**Note: Currently job will post comment on PR with my account (@zahid-arbisoft). We will use any bot account later, it does not need any code change.**

### CODE REVIEW

**Comment by zahid-arbisoft** [comment]
Test

**Comment by zahid-arbisoft** [comment]
$report

**Comment by zahid-arbisoft** [comment]
abc

**Comment by zahid-arbisoft** [comment]
test

**Comment by zahid-arbisoft** [comment]
$body

**Comment by zahid-arbisoft** [comment]
-------------
Diff Coverage
Diff: origin/master...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------

**Comment by zahid-arbisoft** [comment]
-------------
Diff Coverage
Diff: origin/master...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------

**Comment by zahid-arbisoft** [comment]
```
-------------
Diff Coverage
Diff: origin/master...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
------------- 
Check-out [tests](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/289/workflows/08b30f6a-6b9d-4eb0-8fe7-922c5b759e3d/jobs/297/tests) and coverage [reports](artifacts)
```

**Comment by zahid-arbisoft** [comment]
```
## Diff Coverage Report
Check-out complete **[report](artifacts)** and other artifacts.

-------------
Diff Coverage
Diff: origin/master...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report
Check-out complete **[report](artifacts)** and other artifacts.
```
-------------
Diff Coverage
Diff: origin/master...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
This test message

**Comment by zahid-arbisoft** [comment]
This test message

**Comment by zahid-arbisoft** [comment]
This test message

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report
Check-out complete **[report](link)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/318/workflows/workflows_id/jobs/318/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/319/workflows/workflows_id/jobs/319/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/320/workflows/workflows_id/jobs/320/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/324/workflows/workflows_id/jobs/324/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/334/workflows/workflows_id/jobs/334/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/336/workflows/workflows_id/jobs/336/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - Edx
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/340/workflows/workflows_id/jobs/340/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
conftest.py (100%)
-------------
Total:   8 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - Edx
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/342/workflows/workflows_id/jobs/342/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
cms/djangoapps/pipeline_js/test_my_functinon_ok.py (100%)
cms/djangoapps/pipeline_js/utils.py (100%)
conftest.py (100%)
lms/djangoapps/commerce/utils.py (20.0%): Missing lines 421-424
lms/djangoapps/survey/tests/test_utils.py (100%)
lms/djangoapps/survey/tests/test_views.py (100%)
lms/djangoapps/survey/utils.py (100%)
openedx/adg/lms/applications/utils.py (0.0%): Missing lines 1-2,4-6,9,20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71,74-77
pavelib/utils/envs.py (0.0%): Missing lines 224
pavelib/utils/test/suites/pytest_suite.py (0.0%): Missing lines 252-255,257
scripts/xsslint/tests/test_utils.py (100%)
scripts/xsslint/xsslint/utils.py (100%)
-------------
Total:   91 lines
Missing: 61 lines
Coverage: 32%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - Edx
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/344/workflows/workflows_id/jobs/344/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
cms/djangoapps/pipeline_js/test_my_functinon_ok.py (100%)
cms/djangoapps/pipeline_js/utils.py (100%)
conftest.py (100%)
lms/djangoapps/commerce/utils.py (20.0%): Missing lines 421-424
lms/djangoapps/survey/tests/test_utils.py (100%)
lms/djangoapps/survey/utils.py (100%)
openedx/adg/lms/applications/utils.py (0.0%): Missing lines 1-2,4-6,9,20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71,74-77
scripts/xsslint/tests/test_utils.py (100%)
scripts/xsslint/xsslint/utils.py (100%)
-------------
Total:   86 lines
Missing: 55 lines
Coverage: 36%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/345/workflows/workflows_id/jobs/345/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - Edx
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/346/workflows/workflows_id/jobs/346/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
cms/djangoapps/pipeline_js/test_my_functinon_ok.py (100%)
cms/djangoapps/pipeline_js/utils.py (100%)
conftest.py (100%)
lms/djangoapps/commerce/utils.py (20.0%): Missing lines 421-424
lms/djangoapps/survey/tests/test_utils.py (100%)
lms/djangoapps/survey/utils.py (100%)
openedx/adg/lms/applications/utils.py (0.0%): Missing lines 1-2,4-6,9,20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71,74-77
pavelib/utils/envs.py (0.0%): Missing lines 224
pavelib/utils/test/suites/pytest_suite.py (0.0%): Missing lines 252-255,257
scripts/xsslint/tests/test_utils.py (100%)
scripts/xsslint/xsslint/utils.py (100%)
-------------
Total:   92 lines
Missing: 61 lines
Coverage: 33%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/349/workflows/workflows_id/jobs/349/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - Edx
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/350/workflows/workflows_id/jobs/350/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
cms/djangoapps/pipeline_js/test_my_functinon_ok.py (100%)
cms/djangoapps/pipeline_js/utils.py (100%)
conftest.py (100%)
lms/djangoapps/commerce/utils.py (20.0%): Missing lines 421-424
lms/djangoapps/survey/tests/test_utils.py (100%)
lms/djangoapps/survey/utils.py (100%)
openedx/adg/lms/applications/utils.py (0.0%): Missing lines 1-2,4-6,9,20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71,74-77
pavelib/utils/envs.py (0.0%): Missing lines 224
pavelib/utils/test/suites/pytest_suite.py (0.0%): Missing lines 252-255,257
scripts/xsslint/tests/test_utils.py (100%)
scripts/xsslint/xsslint/utils.py (100%)
-------------
Total:   92 lines
Missing: 61 lines
Coverage: 33%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG ((Failed))
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/353/workflows/workflows_id/jobs/353/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/354/workflows/workflows_id/jobs/354/artifacts)**. Coverage !!! Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/358/workflows/workflows_id/jobs/358/artifacts)**. Coverage !!! Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   56 lines
Missing: 41 lines
Coverage: 26%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - Edx ((Failed))
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/352/workflows/workflows_id/jobs/352/artifacts)**.
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
cms/djangoapps/pipeline_js/test_my_functinon_ok.py (100%)
cms/djangoapps/pipeline_js/utils.py (100%)
conftest.py (100%)
lms/djangoapps/commerce/utils.py (20.0%): Missing lines 421-424
lms/djangoapps/survey/tests/test_utils.py (100%)
lms/djangoapps/survey/utils.py (100%)
openedx/adg/lms/applications/utils.py (0.0%): Missing lines 1-2,4-6,9,20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71,74-77
scripts/xsslint/tests/test_utils.py (100%)
scripts/xsslint/xsslint/utils.py (100%)
-------------
Total:   86 lines
Missing: 55 lines
Coverage: 36%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - Edx
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/359/workflows/workflows_id/jobs/359/artifacts)**. Coverage !!! Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
cms/djangoapps/pipeline_js/test_my_functinon_ok.py (100%)
cms/djangoapps/pipeline_js/utils.py (100%)
conftest.py (100%)
lms/djangoapps/commerce/utils.py (20.0%): Missing lines 421-424
lms/djangoapps/survey/tests/test_utils.py (100%)
lms/djangoapps/survey/utils.py (100%)
openedx/adg/lms/applications/utils.py (0.0%): Missing lines 1-2,4-6,9,20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71,74-77
scripts/xsslint/tests/test_utils.py (100%)
scripts/xsslint/xsslint/utils.py (100%)
-------------
Total:   86 lines
Missing: 55 lines
Coverage: 36%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - Edx
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/368/workflows/workflows_id/jobs/368/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/master...HEAD, staged and unstaged changes
-------------
cms/djangoapps/pipeline_js/test_my_functinon_ok.py (100%)
cms/djangoapps/pipeline_js/utils.py (55.6%): Missing lines 14-17
conftest.py (100%)
lms/djangoapps/commerce/utils.py (20.0%): Missing lines 421-424
lms/djangoapps/survey/tests/test_utils.py (100%)
lms/djangoapps/survey/utils.py (100%)
openedx/adg/cms/test_sss.py (100%)
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
pavelib/utils/envs.py (0.0%): Missing lines 224
pavelib/utils/test/suites/pytest_suite.py (0.0%): Missing lines 252-255,257
scripts/xsslint/tests/test_utils.py (100%)
scripts/xsslint/xsslint/utils.py (100%)
-------------
Total:   105 lines
Missing: 55 lines
Coverage: 47%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/369/workflows/workflows_id/jobs/369/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/cms/test_sss.py (100%)
openedx/adg/lms/applications/tests/test_utils.py (100%)
openedx/adg/lms/applications/utils.py (19.6%): Missing lines 20-26,28,30-36,38,40-46,48,50-56,58,61-67,69,71
-------------
Total:   59 lines
Missing: 41 lines
Coverage: 30%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/370/workflows/workflows_id/jobs/370/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
No lines with coverage information in this diff.
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/371/workflows/workflows_id/jobs/371/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
No lines with coverage information in this diff.
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/372/workflows/workflows_id/jobs/372/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
No lines with coverage information in this diff.
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/373/workflows/workflows_id/jobs/373/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
No lines with coverage information in this diff.
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/378/workflows/workflows_id/jobs/378/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
No lines with coverage information in this diff.
-------------
```

**Comment by zahid-arbisoft** [comment]
> **pylint:** ❌
> **pycodestyle:** ✔️
> **iSort:** ❌
> 
> **Please fix following pylint issues:**
> 
> ```
> ************* Module edx-platform.conftest
> conftest.py:23:0: C0116: Missing function or method docstring (missing-function-docstring)
> 
> -----------------------------------
> Your code has been rated at 9.80/10
> ```
> 
> **Please fix following iSort issues:**
> 
> ```
>  ERROR: /conftest.py Imports are incorrectly sorted. 
> ```

These are core changes

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/382/workflows/workflows_id/jobs/382/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/utils/tests/tests_date_utils.py (100%)
-------------
Total:   4 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/383/workflows/workflows_id/jobs/383/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/utils/tests/tests_date_utils.py (100%)
-------------
Total:   4 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/388/workflows/workflows_id/jobs/388/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/utils/tests/tests_date_utils.py (100%)
-------------
Total:   4 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/389/workflows/workflows_id/jobs/389/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/utils/tests/tests_date_utils.py (100%)
-------------
Total:   4 lines
Missing: 0 lines
Coverage: 100%
-------------
```

File: adg/pipelines/scripts/post_comment.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+Python script file for posting coverage report as comment on Github PR
+"""
+# !/usr/bin/env python3
+import argparse
+import os
+
+import requests
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--type', '-t', help='Report type i.e. AAG, Edx')
```
Comment (zeshanarifios) [inline adg/pipelines/scripts/post_comment.py:11]: Please fix typo 
AAG -> ADG

**Review by zeshanarifios — CHANGES_REQUESTED** [review]

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/390/workflows/workflows_id/jobs/390/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/utils/tests/tests_date_utils.py (100%)
-------------
Total:   4 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Review by zeshanarifios — APPROVED** [review]
👍 LGTM

File: adg/pipelines/scripts/post_comment.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    """
+    Post comment on PR using Github api
+    Args:
+        message (str): Coverage Report or comment
+
+    Returns:
+        None
+    """
+    pr_number = CIRCLE_PULL_REQUEST.split('/')[-1]
+    url = "https://api.github.com/repos/{username}/{repo}/issues/{pr}/comments".format(
```
Comment (harris-irfan) [inline adg/pipelines/scripts/post_comment.py:74]: single qoutes

File: adg/pipelines/scripts/post_comment.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        pr=pr_number,
+        username=CIRCLE_PROJECT_USERNAME,
+        repo=CIRCLE_PROJECT_REPONAME,
+    )
+    headers = {'Authorization': 'token {token}'.format(token=GITHUB_BOT_TOKEN)}
+    response = requests.post(url, json={'body': message}, headers=headers)
+    print("Comment status: ", response.status_code, response.reason)
```
Comment (harris-irfan) [inline adg/pipelines/scripts/post_comment.py:81]: single qoutes

File: adg/pipelines/scripts/post_comment.py (hunk 2 above)
Comment (harris-irfan) [inline adg/pipelines/scripts/post_comment.py:74]: single qoutes

**Review by harris-irfan — APPROVED** [review]
Just a couple of comments related to single qoutes usage.

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/393/workflows/workflows_id/jobs/393/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/utils/tests/tests_date_utils.py (100%)
-------------
Total:   4 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Pull request merged**